packages/backend/src/core/activity/processors/StarkexProcessor.ts

```diff
@@ -25,33 +27,52 @@ export function createStarkexProcessor(
   options: StarkexProcessorOptions,
 ): SequenceProcessor {
   const batchSize = getBatchSizeFromCallsPerMinute(options.singleStarkexCPM)
+  const startDay = options.sinceTimestamp.toStartOf('day').toDays()
+  const processRange = async (
+    from: number,
+    to: number,
+    trx?: Knex.Transaction,
+  ) => {
+    const queries = range(from, to + 1).map((day) => async () => {
+      const count = await starkexClient.getDailyCount(day, options.product)
 
-  return new SequenceProcessor(
+      return {
+        count,
+        timestamp: UnixTime.fromDays(day),
+        projectId: projectId,
+      }
+    })
+
+    const counts = await promiseAllPlus(queries, logger)
+    await starkexRepository.addOrUpdateMany(counts, trx)
+  }
+
+  const processor = new SequenceProcessor(
     projectId.toString(),
     logger,
     sequenceProcessorRepository,
     {
       batchSize,
-      startFrom: options.sinceTimestamp.toStartOf('day').toDays(),
+      startFrom: startDay,
       // eslint-disable-next-line @typescript-eslint/require-await
       getLatest: async () =>
         getStarkexLastDay(clock.getLastHour(), options.starkexApiDelayHours),
-      processRange: async (from, to, trx) => {
-        const queries = range(from, to + 1).map((day) => async () => {
-          const count = await starkexClient.getDailyCount(day, options.product)
-
-          return {
-            count,
-            timestamp: UnixTime.fromDays(day),
-            projectId: projectId,
-          }
-        })
-
-        const counts = await promiseAllPlus(queries, logger)
-        await starkexRepository.addOrUpdateMany(counts, trx)
-      },
+      processRange,
     },
   )
+
+  processor.on(ALL_PROCESSED_EVENT, () => {
+    void (async function () {
+      const state = await sequenceProcessorRepository.getById(processor.id)
+      assert(state)
+      await processRange(
+        state.lastProcessed - options.starkexResyncLastDays,
+        state.lastProcessed,
+      )
+    })()
```

Comment:
This has two issues:
1. `void (async () => {})()` reduces readability (I do understand `EventEmitter.on` expects void as a return type though),
2. Async error thrown here is swallowed.

Instead I suggest the following:
```
const handleAllProcessedEvent = async () => {
      const state = await sequenceProcessorRepository.getById(processor.id)
      assert(state)
      await processRange(
        state.lastProcessed - options.starkexResyncLastDays,
        state.lastProcessed,
      )
}

processor.on(ALL_PROCESSED_EVENT, () => {
    handleAllProcessedEvent().catch(logger.error.bind(logger))
})
```

This way we don't fight with type system and get proper error handling.